modules/StateMixin.js

```diff
@@ -0,0 +1,79 @@
+var { isAbsolutePath } = require('./PathUtils');
+
+function routeIsActive(activeRoutes, route) {
+  if (typeof route === 'object')
+    return activeRoutes.indexOf(route) !== -1;
+
+  return activeRoutes.some(function (r) {
+    return r.name === route;
+  });
+}
+
+function paramsAreActive(activeParams, params) {
+  for (var property in params)
+    if (String(activeParams[property]) !== String(params[property]))
+      return false;
+
+  return true;
+}
+
+function queryIsActive(activeQuery, query) {
+  if (activeQuery == null)
+    return false;
+
+  for (var property in query)
+    if (String(activeQuery[property]) !== String(query[property]))
+      return false;
+
+  return true;
+}
+
+var StateMixin = {
```

Comment:
Should probably rename this file so it's not confused with `modules/State` which is also a mixin.